engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -434,7 +434,7 @@ private void validateConstraintsForCurrentGroup(ValidationContext<?> validationC
 
 	private <U> void validateConstraintsForDefaultGroup(ValidationContext<?> validationContext, ValueContext<U, Object> valueContext) {
 		final BeanMetaData<U> beanMetaData = valueContext.getCurrentBeanMetaData();
-		final Map<Class<?>, Class<?>> validatedInterfaces = newHashMap();
+		final Map<Class<?>, Class<?>> validatedInterfaces = new HashMap<>();
```

Comment:
@johnaohara Can you confirm it's the change you had in mind? Thanks!

(I plan to release 6.0.6 today to push it to WF 12)

Reply:
@gsmet yes that was the change for HashMap

Reply:
Interesting, why is it better? But either way it's fine with me :)

Reply:
From what John said, it prevents the compiler from doing optimizations. But apparently, that wasn't expected by John. So we fixed it there for now but it probably requires more investigation on the compiler side.

Reply:
The change allowed ValidatorImpl.validateConstraintsForDefaultGroup() and ValidatorImpl.validateMetaConstraint() to be inlined

Also, there was contention on a futex call from the JVM, changing from a static method call reduced overall the cpu time spent spinning on futexes. This I have not seen before and need to understand better.